File: backend/src/main/java/com/hackathon/service/TeamService.java
package com.hackathon.service;

import com.hackathon.dto.request.TeamRequest;
import com.hackathon.model.Team;

import java.util.List;

public interface TeamService {

    Team create(TeamRequest request, String userId);

    Team joinTeam(String teamId, String userId);

    Team leaveTeam(String teamId, String userId);

    void removeMember(String teamId, String leaderId, String memberId);

    Team getById(String id);

    List<Team> getByHackathonId(String hackathonId);

    List<Team> getMyTeams(String userId);

    Team getMyTeamForHackathon(String userId, String hackathonId);

    void delete(String id, String userId);

    void inviteParticipant(String teamId, String inviterId, String inviteeEmail);

    List<com.hackathon.model.TeamInvitation> getMyInvitations(String userId);

    Team acceptInvitation(String invitationId, String userId);

    void rejectInvitation(String invitationId, String userId);
}

File: frontend/src/api/teamService.js
import api from './axios';

export const teamAPI = {
  create: (data) => api.post('/teams', data),
  join: (teamId) => api.post(`/teams/${teamId}/join`),
  leave: (teamId) => api.post(`/teams/${teamId}/leave`),
  getById: (id) => api.get(`/teams/${id}`),
  getByHackathon: (hackathonId) => api.get(`/teams/hackathon/${hackathonId}`),
  getMyTeams: () => api.get('/teams/my-teams'),
  getMyTeamForHackathon: (hackathonId) => api.get(`/teams/my-team/${hackathonId}`),
  delete: (id) => api.delete(`/teams/${id}`),
  removeMember: (teamId, memberId) => api.delete(`/teams/${teamId}/members/${memberId}`),
  inviteParticipant: (teamId, email) => api.post(`/teams/${teamId}/invite?email=${encodeURIComponent(email)}`),
  getMyInvitations: () => api.get('/teams/invitations'),
  acceptInvitation: (invitationId) => api.post(`/teams/invitations/${invitationId}/accept`),
  rejectInvitation: (invitationId) => api.post(`/teams/invitations/${invitationId}/reject`),
};
